azure-core/azure/core/pipeline/__init__.py

```diff
@@ -45,6 +45,35 @@ def __exit__(self, exc_type, exc_value, traceback):
             return None
 
 
+class PipelineContext(dict):
+
+    def __init__(self, transport, **kwargs):
+        self.transport = transport
+        self.options = kwargs
+        self._protected = ['transport', 'session', 'options']
+
+    def __setitem__(self, key, item):
+        if key in self._protected:
+            raise ValueError('Context value {} cannot be overwritten.'.format(key))
```

Comment:
`session` is a protected attr but its not set in the initializer and can't be set with `__setitem__` either, what sets it?

Reply:
Oops - forgot to remove it from that list - thanks!